### PR TITLE
Assumptions: Symmetric handlers for actual matrices

### PR DESCRIPTION
**Pull request by AnishShah**
While reading the source code, I found a [TODO](https://github.com/sympy/sympy/blob/master/sympy/assumptions/ask.py#L761) tag about adding symmetric handlers for actual matrices. 

Let me know your thoughts. @asmeurer @debugger22

### CODE REVIEW

**Comment by AnishShah** [comment]
Can you review this please? @asmeurer @debugger22

File: sympy/assumptions/handlers/matrices.py [hunk 1]
```diff
@@ -73,6 +74,16 @@ def MatrixSlice(expr, assumptions):
 
     Identity = staticmethod(CommonHandler.AlwaysTrue)
 
+    @staticmethod
+    def ImmutableMatrix(expr, assumptions):
```
Comment (asmeurer) [inline sympy/assumptions/handlers/matrices.py]: Why not MatrixBase?

**Comment by AnishShah** [comment]
I have updated the PR

**Comment by AnishShah** [comment]
@asmeurer The tests are passing now. Any other changes?

**Comment by AnishShah** [comment]
@asmeurer Can this PR get merged?

**Pull request closed without merging**